web/crux-ui/src/providers/websocket.tsx

```diff
@@ -0,0 +1,30 @@
+import { defaultWsErrorHandler } from '@app/errors'
+import { isServerSide } from '@app/utils'
+import { WebSocketClient } from '@app/websockets/client'
+import useTranslation from 'next-translate/useTranslation'
+import React, { useState } from 'react'
+
+interface WebSocketContextInterface {
+  client: WebSocketClient
+}
+
+export const WebSocketContext = React.createContext<WebSocketContextInterface>({ client: null })
+
+export const WebSocketProvider = (props: React.PropsWithChildren<{}>) => {
+  const { t } = useTranslation()
+
+  const [wsClient, _] = useState(() => {
```

Comment:
useRef instead?

Reply:
I wanted to use `useRef` but I need a default value initialiser with a function. `useRef` only supports values.